lib/Settings/Settings.cpp

```diff
@@ -81,3 +87,33 @@ void Settings::updateProtocols(const String &protocols) {
 }
 
 void Settings::updateOtaUrl(const String &otaUrl) { this->otaUrl = otaUrl; }
+
+void Settings::serialize(Stream &stream, bool pretty, bool sensible) {
+  DynamicJsonBuffer jsonBuffer;
+  JsonObject &root = jsonBuffer.createObject();
+
+  root[F("deviceName")] = this->deviceName;
```

Comment:
Here is the reason, why I prefer a `std::map` for the settings. With a `std::map`, this could be a loop. Alternatively we can think about using a JsonObject to store the configurations.

Reply:
See my earlier comments. I see your argument but it causes more complexity in other places in this particular case.